platform-modules/batch-models/src/main/scala/org/ekstep/analytics/job/AssessmentMetricsJob.scala

```diff
@@ -0,0 +1,296 @@
+    val resolvedExternalIdDF = userAssessmentResolvedDF.join(externalIdMapDF, Seq("userid"), "left_outer")
+
+    /*
+    * Resolve organisation name from `rootorgid`
+    * */
+    val resolvedOrgNameDF = resolvedExternalIdDF
+      .join(organisationDF, organisationDF.col("id") === resolvedExternalIdDF.col("rootorgid"), "left_outer")
+      .dropDuplicates(Seq("userid"))
+      .select(resolvedExternalIdDF.col("userid"), col("orgname").as("orgname_resolved"))
+
+
+    /*
+    * Resolve school name from `orgid`
+    * */
+    val resolvedSchoolNameDF = resolvedExternalIdDF
+      .join(organisationDF, organisationDF.col("id") === resolvedExternalIdDF.col("organisationid"), "left_outer")
+      .dropDuplicates(Seq("userid"))
+      .select(resolvedExternalIdDF.col("userid"), col("orgname").as("schoolname_resolved"))
+
+
+    /*
+    * merge orgName and schoolName based on `userid` and calculate the course progress percentage from `progress` column which is no of content visited/read
+    * */
+
+    resolvedExternalIdDF
+      .join(resolvedSchoolNameDF, Seq("userid"), "left_outer")
+      .join(resolvedOrgNameDF, Seq("userid"), "left_outer")
+  }
+
+  /**
+    * De-norming the assessment report - Adding content name column to the content id
+    *
+    * @return - Assessment denormalised dataframe
+    */
+  def denormAssessment(spark: SparkSession, report: DataFrame): DataFrame = {
+    val contentIds = report.select(col("content_id")).rdd.map(r => r.getString(0)).collect.toList.distinct.filter(_ != null)
+    val contentNameDF = AssessmentReportUtil.getContentNames(spark, contentIds)
+    report.join(contentNameDF, report.col("content_id") === contentNameDF.col("identifier"), "right_outer")
+      .select(col("name"),
+        col("total_sum_score"), report.col("userid"), report.col("courseid"), report.col("batchid"),
+        col("total_score"), report.col("maskedemail"), report.col("district_name"), report.col("maskedphone"),
+        report.col("orgname_resolved"), report.col("externalid"), report.col("schoolname_resolved"), report.col("username")
+      )
+  }
+
+
+  /**
+    * This method is used to upload the report the azure cloud service and
+    * Index report data into elastic search.
+    * Alias name: cbatch-assessment
+    * Index name: cbatch-assessment-24-08-1993-09-30 (dd-mm-yyyy-hh-mm)
+    */
+  def saveReport(reportDF: DataFrame, url: String): Unit = {
+    // Save assessment report to ealstic search
+    val alias_name = AppConf.getConfig("assessment.metrics.es.alias")
+    val index_prefix = AppConf.getConfig("assessment.metrics.es.index.prefix")
+    val index_name = AssessmentReportUtil.suffixDate(index_prefix)
+    AssessmentReportUtil.saveToElastic(index_name, alias_name, reportDF)
```

Comment:
Add config to save report:
1. uploadreportEnabled flag
2. indexElasticsearchEnabled flag

Reply:
Added both the configuration before uploading repots to azure and saving into es index